engine/runtime/src/main/java/org/enso/interpreter/epb/node/ContextFlipNode.java

```diff
@@ -0,0 +1,68 @@
+package org.enso.interpreter.epb.node;
+
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.GenerateUncached;
+import com.oracle.truffle.api.dsl.ReportPolymorphism;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.epb.runtime.GuardedTruffleContext;
+import org.enso.interpreter.epb.runtime.PolyglotProxy;
+
+@GenerateUncached
+@ReportPolymorphism
+public abstract class ContextFlipNode extends Node {
```

Comment:
I'm not sure the name of this node really expresses what it does. `ContextProxyNode`?